geowebcache/core/src/main/java/org/geowebcache/seed/TruncateTask.java

```diff
@@ -49,11 +56,40 @@ public TruncateTask(StorageBroker sb, TileRange tr, TileLayer tl, boolean doFilt
     protected void doActionInternal() throws GeoWebCacheException, InterruptedException {
         super.state = GWCTask.STATE.RUNNING;
         checkInterrupted();
-        try {
-            storageBroker.delete(tr);
-        } catch (Exception e) {
-            super.state = GWCTask.STATE.DEAD;
-            log.error("During truncate request", e);
+
+        final String layerName = tl.getName();
+        TileRange tr = trIter.getTileRange();
+        checkInterrupted();
+
+        long[] gridLoc = trIter.nextMetaGridLocation(new long[3]);
+
+        while (gridLoc != null && !this.terminate) {
+
+            checkInterrupted();
+            Map<String, String> fullParameters = tr.getParameters();
+
+            ConveyorTile tile =
+                    new ConveyorTile(
+                            storageBroker,
+                            layerName,
+                            tr.getGridSetId(),
+                            gridLoc,
+                            tr.getMimeType(),
+                            fullParameters,
+                            null,
+                            null);
+
+            try {
+                checkInterrupted();
+                storageBroker.delete(tile.getStorageObject());
+            } catch (Exception e) {
+                super.state = GWCTask.STATE.DEAD;
+                log.error("During truncate request", e);
+                throw new GeoWebCacheException(e);
+            }
+
+            checkInterrupted();
+            gridLoc = trIter.nextMetaGridLocation(gridLoc);
```

Comment:
Going over meta grid locations means deleting only one tile for each metatile, leaving all others intact... or not?